### DISCUSSION TITLE
Give back negative due when overpaid

### DISCUSSION THREAD

**Discussion by JudithWebdev** [btcpayserver/btcpayserver#2900] [General]
When getting the invoice payment methods it is quite handy to know the due. 

Result I would expect:
- When paid correctly: due= 0
- When underpaid: due=1234
- When overpaid: due = -1234

Actual result:
- When paid correctly: due= 0
- When underpaid: due=1234
- When overpaid: due = 0

Wouldn't it be nice if due can be negative?

**Comment by NicolasDorier** [comment]
We can't change overpaid as some people may already depends on `overpaid == 0` for some logic.
Exactly which route API you are talking about?

**Reply by woutersamaey** [reply]
This is about the `due` field in the "Get invoice Payment Methods" API call response.
@Kukks thought it would be fine to return a negative amount.

If we don't want to break BC, we could always add a new field that can be negative while keeping the original as is? Not so pretty, but it's an option.

**Reply by NicolasDorier** [reply]
https://github.com/btcpayserver/btcpayserver/pull/2951
